pkgs/development/libraries/glibc/nss-path.patch

```diff
@@ -0,0 +1,50 @@
+diff -ru -x '*~' glibc-2.32-orig/nss/nsswitch.c glibc-2.32/nss/nsswitch.c
+--- glibc-2.32-orig/nss/nsswitch.c	2020-08-05 04:17:00.000000000 +0200
++++ glibc-2.32/nss/nsswitch.c	2021-01-29 19:40:50.730883061 +0100
+@@ -349,6 +349,35 @@
+ 		__nss_shlib_revision);
+ 
+       ni->library->lib_handle = __libc_dlopen (shlib_name);
++
++      if (ni->library->lib_handle == NULL)
++	{
++	  const char *nss_path = __libc_secure_getenv ("NIX_GLIBC_NSS_PATH");
++	  if (!nss_path)
++	    nss_path = DEFAULT_NSS_PATH;
++
++	  const char *pos = nss_path;
++
++	  while (*pos)
++	    {
++	      const char *end = __strchrnul(pos, ':');
++	      if (pos != end)
++		{
++		  char shlib_path[1024];
++		  size_t shlib_pathlen = (end - pos) + 1 + strlen (shlib_name) + 1;
++		  if (shlib_pathlen < sizeof(shlib_path))
++		    {
++		      __stpcpy (__stpcpy (__stpncpy (shlib_path, pos, end - pos), "/"), shlib_name);
```

Comment:
Is memcpy not better here given that we already calulate the string length?

Reply:
Alternative would be good old `snprintf`...

Reply:
I'm just following the surrounding code here, didn't want to deviate from the coding style.